lib/greed/turn.rb

```diff
@@ -0,0 +1,63 @@
+absolute_path = File.expand_path(File.dirname(__FILE__))
+require absolute_path + '/dice'
+require absolute_path + '/score'
+require 'pry'
+
+module Greed
+  class Turn
+
+    def initialize
+      @dice = Dice.new
+      @score = Score.new
+    end
+
+    # Run a full single turn for a player until they choose to stop, or score 0
+    def take_turn
+      score = 0
+      # Make the initial roll
+      roll(5)
+      result = @score.calc_score(@dice.values)
+      puts "Your initial roll is: #{@dice.values}."
+
+      # While score is not zero:
+      while result['score'] != 0
+        score += result['score']
+
+        # Check if the player wants to roll again
+        break unless reroll?(result['live_dice'], score)
+
+        # Roll the dice
+        roll(result['live_dice'])
+        result = @score.calc_score(@dice.values)
+        puts "You rolled: #{@dice.values} and scored: #{result['score']}."
+      end
+
+      # Return the total score, or zero if the player rolled a zero round
+      return (result['score'] != 0 ? score : 0)
```

Comment:
You would return the result of a single (most recent) roll, instead of an entire round.
The player has the choice to continue rolling 'live' dice (dice that did not score) until they are satisfied, or they roll a zero score. The sum of all their rolls is used if they quit before the 'bust' on a zero. If all rolled dice score, then all five dice are reset. `score` contains the round total, `result['score']` contains only the result of the most recent roll.

From the rules:

> After a player rolls and the score is calculated, the scoring dice are
> removed and the player has the option of rolling again using only the
> non-scoring dice. If all of the thrown dice are scoring, then the
> player may roll all 5 dice in the next roll.
> 
> The player may continue to roll as long as each roll scores points. If
> a roll has zero points, then the player loses not only their turn, but
> also accumulated score for that turn. If a player decides to stop
> rolling before rolling a zero-point roll, then the accumulated points
> for the turn is added to his total score.